scripts/unreleased-change-monitor.ts

```diff
@@ -0,0 +1,124 @@
+import { Octokit } from 'octokit';
+import { LimaAndQemu, AlpineLimaISO } from 'scripts/dependencies/lima';
+import * as tools from 'scripts/dependencies/tools';
+import { WSLDistro, HostResolverHost } from 'scripts/dependencies/wsl';
+import { Dependency, UnreleasedChangeMonitor, getOctokit } from 'scripts/lib/dependencies';
+
+const GITHUB_OWNER = process.env.GITHUB_REPOSITORY?.split('/')[0] || 'rancher-sandbox';
+const GITHUB_REPO = process.env.GITHUB_REPOSITORY?.split('/')[1] || 'rancher-desktop';
+// a (hopefully) unique and communicative key that is used to find issues created by
+// this script by filtering them down to the ones that have it in their title
+const UCMONITOR = 'ucmonitor';
+
+type UnreleasedChangeMonitoringDependency = Dependency & UnreleasedChangeMonitor;
+
+type DependencyState = {
+  dependency: UnreleasedChangeMonitoringDependency,
+  latestReleaseTag: string,
+  hasUnreleasedChanges: boolean,
+};
+
+const dependencies: UnreleasedChangeMonitoringDependency[] = [
+  new LimaAndQemu(),
+  new WSLDistro(),
+  new tools.DockerCLI(),
+  new tools.Steve(),
+  new tools.GuestAgent(),
+  new tools.RancherDashboard(),
+  new AlpineLimaISO(),
+  new HostResolverHost(), // we only need one of HostResolverHost and HostResolverPeer
+];
+
+type Issue = Awaited<ReturnType<Octokit['rest']['search']['issuesAndPullRequests']>>['data']['items'][0];
+
+async function getExistingIssuesFor(dependencyName: string): Promise<Issue[]> {
+  const queryString = `type:issue repo:${ GITHUB_OWNER }/${ GITHUB_REPO } ${ UCMONITOR } in:title ${ dependencyName } in:title sort:updated`;
```

Comment:
I believe `in:title` is a toggle than makes the whole query only operate on the title (so you don't need to include it twice).